### PR TITLE
feat: Adding single apply and clear buttons to apply to all search filters

### PR DESCRIPTION
**Pull request by kristenarmes**
Signed-off-by: Kristen Armes <karmes@lyft.com>

### Summary of Changes

Currently when filtering on the search page, there are `Apply` and `Clear` buttons for every filter section. Only one filter category/value pair could be passed at a time to the search API due to frontend constraints. This change adds just one `Apply filters` button and one `Clear all` button at the bottom of the filters side panel which allows multiple filters to be passed to the API at the same time.

Another change that was made is related to the `Clear all` behavior. The global state was not aware of the filters in the input text fields until they were applied, so with the new UI changes the single clear button did not work at first if the filters had been typed in but not applied. Now the global state is updated whenever anything is typed into the input fields so that the `Clear all` button will work at any time.

### Tests

Since the `Apply filters` and `Clear all` buttons were moved to a higher level component than they were previously a part of, some tests for those were moved to `SearchFilter` where they are now rendered. A test was also added to ensure the filters global state is updated on any input changes.

### Documentation

N/A

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [X] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [X] PR includes a summary of changes.
- [X] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [X] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

File: frontend/amundsen_application/static/js/pages/TableDetailPage/TableIssues/index.tsx [hunk 1]
```diff
@@ -132,9 +139,9 @@ export class TableIssues extends React.Component<TableIssueProps> {
       return reportIssueLink;
     }
 
-    if (openIssuesUrl && closedIssuesUrl) {
-      return (
-        <span className="table-more-issues" key="more-issue-link">
+    return (
```
Comment (kristenarmes) [inline frontend/amundsen_application/static/js/pages/TableDetailPage/TableIssues/index.tsx:142]: The unrelated changes here were to fix a small bug that showed up if a table only had associated issues that were closed and hadn't been updated in over 30 days. The incorrect link was provided in that case and the user could not view the closed issues.

File: frontend/amundsen_application/static/.betterer.results [hunk 2]
```diff
@@ -762,100 +762,21 @@ exports[`eslint`] = {
...
-    ],
-    "js/pages/SearchPage/SearchFilter/index.spec.tsx:3995988813": [
-      [67, 6, 25, "Use object destructuring.", "354229464"],
-      [68, 6, 29, "Use object destructuring.", "2645724888"],
-      [69, 6, 48, "Use array destructuring.", "1915473085"],
-      [70, 6, 45, "Use array destructuring.", "1487312825"],
-      [104, 14, 7, "\'content\' is already declared in the upper scope.", "3716929964"],
-      [119, 6, 25, "Use object destructuring.", "354229464"],
-      [120, 6, 29, "Use object destructuring.", "2645724888"],
-      [144, 6, 29, "Use object destructuring.", "2645724888"]
```
Comment (Golodhros) [inline frontend/amundsen_application/static/.betterer.results:829]: Many many thanks for this!!
❤️

File: frontend/amundsen_application/static/js/ducks/search/filters/sagas.ts [hunk 3]
```diff
@@ -14,18 +14,20 @@ import { UpdateSearchFilter, UpdateFilterRequest } from './reducer';
...
   };
 
-  if (value === undefined) {
-    resourceFilters = filterFromObj(resourceFilters, [categoryId]);
-  } else {
-    resourceFilters[categoryId] = value;
-  }
+  searchFilters.forEach((filter) => {
+    if (filter.value === undefined) {
+      resourceFilters = filterFromObj(resourceFilters, [filter.categoryId]);
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/ducks/search/filters/sagas.ts:26]: Just to make sure, this is not overwriting the previous filters on each of the "searchFilters" items, right?

**Review by Golodhros — APPROVED** [review]
LGTM

Great job!

File: frontend/amundsen_application/static/js/ducks/search/filters/sagas.ts (hunk 3 above)
Comment (kristenarmes) [reply frontend/amundsen_application/static/js/ducks/search/filters/sagas.ts:26]: That's correct, each apply will still retain the filters from the previous search (minus any that may have been cleared) and will just add on any new ones in addition

**Comment by miguel-lobato-smartprotection** [comment]
Hi, after deploying this new feature it is acting strange. Once you access to the search page from the home page by clicking in a tag, search results do not appear (and you have to click apply filters again).

**Comment by kristenarmes** [comment]
@miguel-lobato-smartprotection thanks for letting me know, I'll look into this
